cap-containers/map.h

```diff
@@ -44,6 +44,14 @@ typedef struct cap_map {
 	struct cap_map *_forward[CAP_MAP_MAX_SKIPLIST_SIZE];
 } cap_map;
 
+// Since on the Iterator for cap_map we are traversing through the singly linked list, we do not need to store an array for all elements, We only need a single _current pointer which will point to the current element which we are seeing(initially it will be map->_forward[0]). If the cap_map is empty, the item with index 0 will be NULL, in that case, the _key and _value should also be NULL, we expect the user to to check against NULL.
```

Comment:
We don't need documentation for Abstract Data Types, only write docs for the member functions in JavaDoc style (like other containers). We only need a comment above an ADT only if it's very essential like a very specific design decision. Please take the explanatory comments to JavaDoc documentation of the member functions.